FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -116,7 +124,44 @@ extension ModelDownloadTask: URLSessionDownloadDelegate {
     }
 
     guard (200 ..< 299).contains(response.statusCode) else {
-      // TODO: Handle download url expiry + retries.
+      /// Possible failure due to download URL expiry.
+      if response.statusCode == 400 {
+        let currentDateTime = Date()
+        if currentDateTime > remoteModelInfo.urlExpiryTime {
```

Comment:
For readability in the else statement and less indenting (I had to carefully measure where this `else` ended), I wonder if a `guard` statement would be better here too? Or can you not have a nested guard? Haven't tried that before.

```
guard remoteModelInfo.urlExpiryTime > currentDateTime else {
  downloadStatus = .failed
  downloadHandlers.completion(.......)
  return
}

// URL has not expired, try the download again.
```

Reply:
Good call, changed.